#Config file for configuring autonomous modes and actions

# MODES ---------------------------------------------------------------------------

# ALLIANCE will be replaced with whatever alliance we are "red" or "blue"



auto_mode_1: ["four_note_source1_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_source2_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_source3_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_source4_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_source5_ALLIANCE_csv"]

auto_mode_2: ["four_note_source1_ALLIANCE_csv", "pause_half", 
              "four_note_source2_ALLIANCE_csv", "pause_half", 
              "four_note_source3_ALLIANCE_csv", "pause_half", 
              "four_note_source4_ALLIANCE_csv", 
              "four_note_source5_ALLIANCE_csv"]

auto_mode_3: ["four_note_speaker1_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_speaker2_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_speaker3_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_speaker4_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_speaker5_ALLIANCE_csv", "pause_half", "relocalize", 
              "four_note_speaker6_ALLIANCE_csv"]

auto_mode_4: ["four_note_speaker1_ALLIANCE_csv", "pause_half", 
              "four_note_speaker2_ALLIANCE_csv", "pause_half", 
              "four_note_speaker3_ALLIANCE_csv", "pause_half", 
              "four_note_speaker4_ALLIANCE_csv", 
              "four_note_speaker5_ALLIANCE_csv", 
              "four_note_speaker6_ALLIANCE_csv"]

auto_mode_5: ["pause_half"]
auto_mode_6: ["pause_half"]
auto_mode_7: ["pause_half"]
auto_mode_8: ["pause_half"]
auto_mode_9: ["pause_half"]

#auto_mode_4_blue: ["map_rel_four_note_grid_right1_blue_csv", "pause_half", "relocalize", "map_rel_four_note_grid_right2_blue_csv", "pause_half", "relocalize", "map_rel_four_note_grid_right3_blue_csv", "pause_half", "relocalize", "map_rel_four_note_grid_right4_blue_csv"]
#auto_mode_4_red: ["map_rel_four_note_grid_right1_red_csv", "pause_half", "relocalize", "map_rel_four_note_grid_right2_red_csv", "pause_half", "relocalize", "map_rel_four_note_grid_right3_red_csv", "pause_half", "relocalize", "map_rel_four_note_grid_right4_red_csv"]
#auto_mode_5: ["map_rel_four_note_grid1_csv", "map_rel_four_note_grid2_csv", "map_rel_four_note_grid3_csv", "map_rel_four_note_grid4_csv"]
#auto_mode_6: ["relocalize", "map_rel_four_note_grid1_csv", "relocalize", "map_rel_four_note_grid2_csv", "relocalize", "map_rel_four_note_grid3_csv", "relocalize", "map_rel_four_note_grid4_csv"]

# auto_mode_7_blue: ["map_rel_four_note_grid_right1_blue_csv", "map_rel_four_note_grid_right2_blue_csv", "map_rel_four_note_grid_right3_blue_csv", "map_rel_four_note_grid_right4_blue_csv"]
# auto_mode_7_red: ["map_rel_four_note_grid_right1_red_csv", "map_rel_four_note_grid_right2_red_csv", "map_rel_four_note_grid_right3_red_csv", "map_rel_four_note_grid_right4_red_csv"]
# auto_mode_8: ["align_to_speaker", "map_rel_four_note1_csv", "align_to_speaker", "map_rel_four_note2_csv", "align_to_speaker", "map_rel_four_note3_csv", "align_to_speaker"]
# auto_mode_9: ["align_to_speaker", "relocalize", "map_rel_four_note1_csv", "align_to_speaker", "relocalize", "map_rel_four_note2_csv", "align_to_speaker", "relocalize", "map_rel_four_note3_csv", "align_to_speaker"]

# ACTIONS ------------------------------------------------------------------------
# must have a value for "type", other stuff varies

pause_half:
  type: pause
  duration: 0.5

align_to_speaker:
  type: align_to_speaker


ros_rate: 10 # in ms, used for all ros rate calls
angle_tolerance: 0.00872665 # 0.5 degrees

relocalize:
    type: "relocalize"
    
snap_180_before_intake:
    type: snap_to_orientation
    angle: 3.14159268 # radians, and i love sig figs

snap_0_before_intake:
    type: snap_to_orientation
    angle: 0.0

cmd_vel_1:
    type: cmd_vel
    duration: 2.75
    cmd_vel:
        x: 1.0 #1.0
        y: -0.4 # -0.4
        z: 0.0
